### PR TITLE
Add model, controller and view tests for comments

### PR DESCRIPTION
**Pull request by IraVolyk**
1. Create model spec
2. Create controller spec [ index, edit, create, update, destroy] 
3. Create view spec [ _comment,  _form, _edit, _index]
4. Add gems:  'factory_bot_rails',  'rspec-rails',  'shoulda-matchers', 'faker',  'rails-controller-testing',  'rspec-html-matchers'

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -56,6 +63,7 @@ def find_card!
   end
 
   def comment_params
+    # binding.pry
```
Comment (yuravenzhik) [inline app/controllers/comments_controller.rb:66]: remove

File: spec/models/comment_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    it { is_expected.to belong_to(:card) }
+  end
+
+  describe 'validations' do
+    subject { create(:comment) }
+
+    context "valid length" do   
+      it "has no errors" do
+        expect(subject.valid?).to be true
+        expect(subject.errors.to_a).to be_empty
```
Comment (yuravenzhik) [inline spec/models/comment_spec.rb:16]: you may not check because it is checked in `valid?`

File: spec/models/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+RSpec.describe Comment, type: :model do
+ 
+  describe 'associations' do
+    it { is_expected.to belong_to(:user) }
+    it { is_expected.to belong_to(:card) }
+  end
+
+  describe 'validations' do
+    subject { create(:comment) }
```
Comment (yuravenzhik) [inline spec/models/comment_spec.rb:11]: it may not be subject
I guess subject there is `user.valid?`

File: spec/models/comment_spec.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    context "valid length" do   
+      it "has no errors" do
+        expect(subject.valid?).to be true
+        expect(subject.errors.to_a).to be_empty
+      end
+    end
+
+    context "invalid length" do
+      it "has related errors" do
+        subject.body = "b" * 1001
```
Comment (yuravenzhik) [inline spec/models/comment_spec.rb:22]: use `user` not subject ti make some changes, subject is to call method that you are checking
also move this to before

File: spec/requests/comments_request_spec.rb [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+    let(:user) { create(:user) }
+
+    context "not authorized user" do
+      before do
+        sign_out user
+      end
+
+      it "has status unauthorized" do
+        subject
+        expect(response.status).to eq(401)
```
Comment (yuravenzhik) [inline spec/requests/comments_request_spec.rb:25]: check error message too

File: spec/requests/comments_request_spec.rb [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+        expect(assigns(:comments)).to match_array(card_comments)
+        expect(assigns(:comments)).to_not contain_exactly(random_comments)
+      end
+    end
+
+    context "with pagination" do
+      it 'does paginate records' do
+        create_list(:comment, 7, card: card, user: user)
+        expected_comments = Comment.order(created_at: :asc).limit(2)
+        get board_column_card_comments_path(board, column, card, page: 2, format: :js), xhr: true
```
Comment (yuravenzhik) [inline spec/requests/comments_request_spec.rb:63]: move to `subject`, and code above to `before`

File: spec/requests/comments_request_spec.rb [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+  describe "#create" do
+    subject { post board_column_card_comments_path(board, column, card, comment: comment_params, format: :js) }
+    let(:card) { create (:card) }
+    let(:column) { card.column }
+    let(:board) { column.board }
+    let(:user) { create(:user) }
+
+    context "not authorized user" do
+      let(:comment_params) do
+        {body: "Test"}
```
Comment (yuravenzhik) [inline spec/requests/comments_request_spec.rb:101]: check code style

File: spec/requests/comments_request_spec.rb [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+
+      it "has status unauthorized" do
+        subject
+        expect(response.status).to eq(401)
+      end
+    end
+
+    context "correct params are passed" do
+      let(:comment_params) do
+        {body: "Test"}
```
Comment (yuravenzhik) [inline spec/requests/comments_request_spec.rb:116]: code style

File: spec/requests/comments_request_spec.rb [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+      end
+
+      before do
+        sign_out user
+      end
+
+      it "has status unauthorized" do
+        subject
+        expect(response.status).to eq(401)
```
Comment (yuravenzhik) [inline spec/requests/comments_request_spec.rb:110]: check error, and check tests below

File: spec/requests/comments_request_spec.rb (hunk 8 above)
Comment (yuravenzhik) [reply spec/requests/comments_request_spec.rb:116]: also check tests below

File: spec/requests/comments_request_spec.rb [hunk 10]
```diff
@@ -0,0 +1,310 @@
...
+      end
+
+      it "has correct user" do
+        subject
+        expect(Comment.last.user).to eq user 
+      end
+
+      it "has correct body" do
+        subject
+        expect(Comment.last.body).to eq comment_params[:body]
```
Comment (yuravenzhik) [inline spec/requests/comments_request_spec.rb:146]: this and previous tests may be combined in 1

File: spec/models/comment_spec.rb (hunk 3 above)
Comment (IraVolyk) [reply spec/models/comment_spec.rb:11]: subject { comment.valid? }
let(:comment) { create(:comment) }

**Pull request merged**